src/classes/UTIL_DMLService_TEST.cls

```diff
@@ -0,0 +1,699 @@
+                                                        new Contact(FirstName = 'Astro', LastName = 'Cloud', AccountId = acct.Id) };
+                         
+        Test.startTest();
+        UTIL_DMLService.insertRecords(contacts);
+        Test.stopTest();
+
+        List<Contact> retrievedContacts = [SELECT Id, FirstName, LastName, AccountId 
+                                           FROM Contact];
+
+        if (!retrievedContacts.isEmpty()) {
+            System.assertEquals(contacts[0].FirstName, retrievedContacts[0].FirstName);
+            System.assertEquals(contacts[1].FirstName, retrievedContacts[1].FirstName);
+        }
+    }
+
+
+    /*********************************************************************************************************
+    * @description Tests insert of a record with specified optAllOrNone value and default DMLOptions settings.
+    * verify:
+    *   Record inserted
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRecordInsertWithAllOrNone() {
+        if (strTestOnly != '*' && strTestOnly != 'testRecordInsertWithAllOrNone') return;
+        
+        Account acct = new Account(Name = 'DMLOptionsAccount');
+        insert acct;
+
+        Contact con = new Contact(FirstName = 'Blue', LastName = 'Cloud', AccountId = acct.Id); 
+                 
+        Test.startTest();
+        UTIL_DMLService.insertRecord(con, false);
+        Test.stopTest();
+
+        List<Contact> retrievedContacts = [SELECT Id, FirstName, LastName, AccountId 
+                                           FROM Contact];
+
+        if (!retrievedContacts.isEmpty()) {
+            System.assertEquals(con.FirstName, retrievedContacts[0].FirstName);
+        }
+    }
+
+
+    /*********************************************************************************************************
+    * @description Tests insert of a List of records with specified optAllOrNone value and default DMLOptions settings.
+    * verify:
+    *   Records inserted
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRecordsInsertWithAllOrNone() {
+        if (strTestOnly != '*' && strTestOnly != 'testRecordsInsertWithAllOrNone') return;
+        
+        Account acct = new Account(Name = 'DMLOptionsAccount');
+        insert acct;
+
+        List<Contact> contacts = new List<Contact> { new Contact(FirstName = 'Blue', LastName = 'Cloud', AccountId = acct.Id),
+                                                        new Contact(FirstName = 'Astro', LastName = 'Cloud', AccountId = acct.Id) };
+                         
+        Test.startTest();
+        UTIL_DMLService.insertRecords(contacts, false);
```

Comment:
if the goal is to really ensure your cover correctly handles the allOrNone flag, then I'd expect to see a test that inserts two contacts, one without a lastname (which should get an error).  if allorNone=true, then 0 contacts saved.  if allOrNone=false, then 1 contact saved.

Reply:
Sounds good.